service/dependency.go

```diff
@@ -49,3 +53,21 @@ func (s *Service) DependenciesFromService() []*DependencyFromService {
 	}
 	return d
 }
+
+// Logs gives the dependency logs. rstd stands for standard logs and rerr stands for
+// error logs.
+func (d *DependencyFromService) Logs() (rstd, rerr io.ReadCloser, err error) {
+	var reader io.ReadCloser
+	reader, err = defaultContainer.ServiceLogs(d.namespace())
+	if err != nil {
+		return nil, nil, err
+	}
+	sr, sw := io.Pipe()
+	er, ew := io.Pipe()
+	go func() {
```

Comment:
make sure the go routine is finished when connection is closed

Reply:
_StdCopy_ returns when there is an error writing to _sw_ or _ew_ which will be caused by calling _Close()_ on _sr_ or _er_.

so yes, goroutine finishes when one of the readers is closed